crates/stackable-operator/src/utils/cluster_domain.rs

```diff
@@ -0,0 +1,183 @@
+/// - <https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/>
+pub static KUBERNETES_CLUSTER_DOMAIN: OnceLock<DomainName> = OnceLock::new();
+
+#[instrument]
+pub(crate) fn retrieve_cluster_domain() -> Result<DomainName, Error> {
+    // 1. Read KUBERNETES_CLUSTER_DOMAIN env var
+    tracing::debug!("Trying to determine the Kubernetes cluster domain...");
+
+    match env::var(KUBERNETES_CLUSTER_DOMAIN_ENV) {
+        Ok(cluster_domain) if !cluster_domain.is_empty() => {
+            let cluster_domain = DomainName::from_str(&cluster_domain)
+                .context(ParseDomainNameSnafu { cluster_domain })?;
+            tracing::info!(
+                %cluster_domain,
+                "Using Kubernetes cluster domain from {KUBERNETES_CLUSTER_DOMAIN_ENV} environment variable"
+            );
+            return Ok(cluster_domain);
+        }
+        _ => {}
+    };
+
+    // 2. If no env var is set, check if we run in a clustered (Kubernetes/Openshift) environment
+    //    by checking if KUBERNETES_SERVICE_HOST is set: If not default to 'cluster.local'.
+    tracing::debug!(
+        "Trying to determine the operator runtime environment as environment variable \
+            \"{KUBERNETES_CLUSTER_DOMAIN_ENV}\" is not set"
+    );
+
+    match env::var(KUBERNETES_SERVICE_HOST_ENV) {
+        Ok(_) => {
+            let cluster_domain = retrieve_cluster_domain_from_resolv_conf(RESOLVE_CONF_FILE_PATH)?;
+            let cluster_domain = DomainName::from_str(&cluster_domain)
+                .context(ParseDomainNameSnafu { cluster_domain })?;
+
+            tracing::info!(
+                %cluster_domain,
+                "Using Kubernetes cluster domain from {RESOLVE_CONF_FILE_PATH} file"
+            );
+
+            Ok(cluster_domain)
+        }
+        Err(_) => {
+            let cluster_domain = DomainName::from_str(KUBERNETES_CLUSTER_DOMAIN_DEFAULT).context(
+                ParseDomainNameSnafu {
+                    cluster_domain: KUBERNETES_CLUSTER_DOMAIN_DEFAULT,
+                },
+            )?;
+
+            tracing::info!(
+                %cluster_domain,
+                "Could not determine Kubernetes cluster domain as the operator is not running within Kubernetes, assuming default Kubernetes cluster domain"
+            );
+            Ok(cluster_domain)
+        }
+    }
+}
+
+#[instrument]
+fn retrieve_cluster_domain_from_resolv_conf<P>(path: P) -> Result<String, Error>
+where
+    P: std::fmt::Debug + AsRef<Path>,
+{
+    let content = std::fs::read_to_string(path).context(ReadResolvConfFileSnafu)?;
+
+    let last = content
+        .lines()
+        .map(|l| l.trim())
+        .filter(|l| l.starts_with("search"))
+        .map(|l| l.trim_start_matches("search"))
+        .last()
+        .context(NoSearchEntrySnafu)?;
+
+    let shortest_entry = last
+        .split_ascii_whitespace()
+        .min_by_key(|item| item.len())
+        .context(AmbiguousDomainEntriesSnafu)?;
```

Comment:
suggestion: Emit a warning and fall back to default cluster domain.

```suggestion
    let Ok(content) = std::fs::read_to_string(path) else {
        // todo: use PathBuf to display, as the below won't work
        tracing::warn!(path, "Cannot read resolv conf. Falling back to default cluster domain {KUBERNETES_CLUSTER_DOMAIN_DEFAULT:?}");
        return Ok(KUBERNETES_CLUSTER_DOMAIN_DEFAULT)
    };
    
    let Some(last) = content
        .lines()
        .map(|l| l.trim())
        .filter(|l| l.starts_with("search"))
        .map(|l| l.trim_start_matches("search"))
        .last()
    else {
        tracing::warn!(path, "Cannot find search entry. Falling back to default cluster domain {KUBERNETES_CLUSTER_DOMAIN_DEFAULT:?}");
        return Ok(KUBERNETES_CLUSTER_DOMAIN_DEFAULT)
    };

    let Some(shortest_entry) = last
        .split_ascii_whitespace()
        .min_by_key(|item| item.len())
    else {
        tracing::warn!(path, "Ambiguous search suffixes. Falling back to default cluster domain {KUBERNETES_CLUSTER_DOMAIN_DEFAULT:?}");
        return Ok(KUBERNETES_CLUSTER_DOMAIN_DEFAULT)
    };
```

Reply:
Done in c6865684dababfdb643dfbf2a5d062aa38514822